src/main/java/com/mparticle/kits/OneTrustKit.java

```diff
@@ -12,16 +32,90 @@
     private final static String MP_MOBILE_CONSENT_GROUPS = "mobileConsentGroups";
     private final static String ONETRUST_PREFS = "OT_mP_Mapping";
     private Context m_context;
-    
+
+    private BroadcastReceiver categoryReceiver;
+
     @Override
-    protected List<ReportingMessage> onKitCreate(Map<String, String> settings, Context context) {
+    protected List<ReportingMessage> onKitCreate(Map<String, String> settings, final Context context) {
         // Retrieve mParticle --> OneTrust mapping values
         String mobileMappingValues = settings.get(MP_MOBILE_CONSENT_GROUPS);
+
+        final Map<String, String> consentMapping = new HashMap<String, String>();
+
+        if (mobileMappingValues != null) {
+            try {
+                // Convert mobileMappingValues and extract new consentMapping
+                // value -> OneTrust Cookie Value
+                // map -> mParticle Consent Purpose
+                JSONArray consentJSONArray = new JSONArray(mobileMappingValues);
+                for (int i = 0; i < consentJSONArray.length(); i++) {
+                    JSONObject consentJSONObject = consentJSONArray.optJSONObject(i);
+
+                    consentMapping.put(consentJSONObject.getString("value"), consentJSONObject.getString("map"));
```

Comment:
`JSONObject` is a bit tricky...if the field isn't present, calling `getString()` will throw an exception and, in this case, short-circut the entire for-loop.

Either we need to call `optString()` and check the return value with `MPUtility.isEmpty()` OR we can add a second try/catch around just this line, that way if the field is missing, then we can ignore it and move on to the next JSONObject